handlers/search.go

```diff
@@ -267,7 +265,42 @@ func SearchAhead(w http.ResponseWriter, r *http.Request) {
 				WHERE from_address_text LIKE $1 || '%'
 			) a 
 			GROUP BY from_address_text`, lowerStrippedSearch)
-
+	case "count_indexed_validators_by_withdrawal_credential":
+		var ensData *types.EnsDomainResponse
+		if utils.IsValidEnsDomain(search) {
+			ensData, _ = GetEnsDomain(search)
+			if len(ensData.Address) > 0 {
+				lowerStrippedSearch = strings.ToLower(strings.Replace(ensData.Address, "0x", "", -1))
+			}
+		}
+		if !searchLikeRE.MatchString(lowerStrippedSearch) {
+			break
+		}
+		const beginningOfSetCredentials string = "010000000000000000000000"
+		if len(lowerStrippedSearch) <= 40 && lowerStrippedSearch[:24] != beginningOfSetCredentials {
+			lowerStrippedSearch = beginningOfSetCredentials + lowerStrippedSearch
+		}
+		// find validators per withdrawal credential (limit result by N addresses and M validators per credential)
+		dbFinding := []struct {
+			DecodedCredential []byte `db:"withdrawalcredentials"`
+			Count             uint64 `db:"count"`
+		}{}
+		err = db.ReaderDb.Select(&dbFinding, `
+			SELECT withdrawalcredentials, COUNT(*) FROM validators
+			WHERE ENCODE(withdrawalcredentials, 'hex') LIKE $1 || '%'
+			GROUP BY withdrawalcredentials`, lowerStrippedSearch)
+		if err == nil {
+			res := make([]struct {
+				EncodedCredential string `json:"withdrawalcredentials"`
+				Count             uint64 `json:"count"`
+			}, len(dbFinding))
+			for i := range dbFinding {
+				res[i].EncodedCredential = fmt.Sprintf("%x", dbFinding[i].DecodedCredential)
+				res[i].Count = dbFinding[i].Count
+				fmt.Println(res[i].EncodedCredential)
```

Comment:
**issue:** Why print the credentials?

Reply:
Oops! it was a rest from a debugging session as you probably guessed it.